x/oracle/keeper/keeper.go

```diff
@@ -98,13 +98,13 @@ func NewKeeper(
 }
 
 // RemoveCurrencyPair removes a given CurrencyPair from state, i.e. removes its nonce + QuotePrice from the module's store.
-func (k Keeper) RemoveCurrencyPair(ctx sdk.Context, cp types.CurrencyPair) {
-	k.currencyPairs.Remove(ctx, cp.ID())
+func (k Keeper) RemoveCurrencyPair(ctx sdk.Context, cpID string) {
+	k.currencyPairs.Remove(ctx, cpID)
 }
 
 // HasCurrencyPair returns true if a given CurrencyPair is stored in state, false otherwise.
-func (k Keeper) HasCurrencyPair(ctx sdk.Context, cp types.CurrencyPair) bool {
-	ok, err := k.currencyPairs.Has(ctx, cp.ID())
+func (k Keeper) HasCurrencyPair(ctx sdk.Context, cpID string) bool {
+	ok, err := k.currencyPairs.Has(ctx, cpID)
```

Comment:
Moving `keeper` methods to use `cpID` instead of a `cp` simplifies the logic for the callers of these functions.  Before, a `cp` would need to be constructed, only for its `String()` or `Ticker()` representation to be used internally